src/controllers/user.controllers.ts

```diff
@@ -0,0 +1,44 @@
+import { Request, Response } from "express";
+import bcrypt from "bcrypt";
+import * as jwt from "jsonwebtoken";
+import dotenv from "dotenv";
+import User from "../models/user";
+import { PayloadAttributes, UserAttributes } from "../types/user.types";
+
+dotenv.config();
+
+const userSignup = async (req: Request, res: Response) => {
+  try {
+    const { firstName, lastName, email, password } = req.body;
+    const salt = await bcrypt.genSalt(12);
+    const hashedPassword: string = await bcrypt.hash(password, salt);
+
+    const newUser: UserAttributes = await User.create({
+      firstName,
+      lastName,
+      email,
+      password: hashedPassword
+    });
+
+    if (newUser) {
+      const payload: PayloadAttributes = {
+        id: newUser.id as string,
+        email: newUser.email as string
+      };
+      const token: string = jwt.sign(
```

Comment:
Create a function that generates the token, cause it might be needed somewhere else.